numpy/lib/function_base.py

```diff
@@ -1925,19 +1941,66 @@ def cov(m, y=None, rowvar=1, bias=0, ddof=None):
     if X.shape[0] == 1:
```

Comment:
Instead of all this, I'd just transpose `X`. In fact, I'd get rid of `X` and reuse `m`.